### PR TITLE
Add hyperparameter tuning

### PR DESCRIPTION
**Pull request by ekallevik**
MVP implementation of parameter tuning. Will add cross-validation when we have decided on the method.

### CODE REVIEW

File: heuristic/state.py [hunk 1]
```diff
@@ -7,7 +7,8 @@
 
 
 class State:
-    def __init__(self, decision_vars, soft_vars, hard_vars, objective_function_value, f):
+    def __init__(self, decision_vars, soft_vars, hard_vars, objective_function_value, f,
+                 hard_penalty=1):
```
Comment (hkgrov) [inline heuristic/state.py:11]: Why do you set the hard_penalties to 1? Wouldn't this be different from our current hard_penalty?

Comment (ekallevik) [reply heuristic/state.py:11]: Yeah, good catch! What is the right value? 12? 10?

Comment (hkgrov) [reply heuristic/state.py:11]: The current hard penalty constant is 12

Comment (ekallevik) [reply heuristic/state.py]: Fixed

**Pull request merged**